sched/environ/env_dup.c

```diff
@@ -61,7 +61,7 @@
  *
  ****************************************************************************/
 
-int env_dup(FAR struct task_group_s *group)
+int env_dup(FAR struct task_group_s *group, FAR char * const *envcp)
 {
   FAR struct tcb_s *ptcb = this_task();
```

Comment:
remove ptcb, let's caller always pass the envcp correctly.

Reply:
It will complicate this a bit, as ptcb also contains envlen. Maybe I have to calculate the envlen like was done with argv (manually). I tried to avoid this.

Reply:
We can pass envlen if you don't want to compute it every time.

Reply:
Of course, but maybe to keep this like argv it should be computed, or I can make a struct that is passed that contains both the pointer and len. Many options how to do this, I'll try to make up my mind.

Reply:
On further thought, can this create some kind of race condition if the parent is killed or the environment is modified ?

This function has sched_lock() which is I assume intended to protect tg_envp.